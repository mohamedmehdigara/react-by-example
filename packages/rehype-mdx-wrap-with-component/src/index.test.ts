import mdxHastToJsx from '@mdx-js/mdx/mdx-hast-to-jsx'
import rehype from 'rehype'
import { Processor } from 'unified'
import rehypeMdxWrapWithComponent from '.'

const fixtures = {
  simple: `
    <h1>Hello world!</h1>

    <!--component Foo-->
    <!--prop bar-->
    <a href="/bar">bar</a>
    <!--prop baz-->
    <pre>code</pre>
    <!--end-->


    <!--component Qux-->
    <!--no prop marker-->
    <p>lorem ipsum</p>
    <!--end-->

    <!--component Quux-->
    <!--prop quuz-->
    <video src="./video.mp4"></video>
    <img src="./img.jpg"></img>
    <label for="input">a label</label>
    <input id="input"></input>
    <!--literal corge="grault"-->
    <!--end-->
  `,
}
describe('rehypeMdxWrapWithComponent', () => {
  let processor: Processor

  beforeEach(() => {
    processor = rehype()
      .use(rehypeMdxWrapWithComponent)
      .use(() => ({ children }) => {
        return {
          type: 'element',
          tagName: '',
          children,
        }
      })
      .use(mdxHastToJsx)
  })

  it.each(Object.entries(fixtures))(
    'should inject nodes "%s"',
    async (_name, html) => {
      const result = await processor.process(html)
      expect(result.toString()).toMatchSnapshot()
    },
  )
})
