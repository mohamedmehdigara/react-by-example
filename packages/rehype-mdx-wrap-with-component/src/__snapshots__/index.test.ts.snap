// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehypeMdxWrapWithComponent should inject nodes "simple" 1`] = `
"<><html><head parentName=\\"html\\"></head><body parentName=\\"html\\"><h1 parentName=\\"body\\">{\`Hello world!\`}</h1>{\`

    \`}
      <Foo
        
          bar={
            <>{\`
    \`}<a {...{\\"href\\":\\"/bar\\"}}>{\`bar\`}</a>{\`
    \`}</>
          }
          
          baz={
            <>{\`
    \`}<pre>{\`code\`}</pre>{\`
    \`}</>
          }
          
      />{\`


    \`}
      <Qux
        
      />{\`

    \`}
      <Quux
        
          quuz={
            <>{\`
    \`}<video {...{\\"src\\":\\"./video.mp4\\"}}></video>{\`
    \`}<img {...{\\"src\\":\\"./img.jpg\\"}}></img>{\`
    \`}<label {...{\\"htmlFor\\":\\"input\\"}}>{\`a label\`}</label>{\`
    \`}<input {...{\\"id\\":\\"input\\"}}></input>{\`
    \`}</>
          }
          
      />{\`
  \`}</body></html></>"
`;
